src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -0,0 +1,37 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.NeutralMode;
+import com.ctre.phoenix6.hardware.TalonFX;
+import com.ctre.phoenix6.signals.NeutralModeValue;
+
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.RobotMap;
+
+public class IntakeSubsystem extends SubsystemBase {
+
+    private final TalonFX intakeMotor;
+
+    private final DigitalInput rightBeamBreak;
+    private final DigitalInput leftBeamBreak;
+
+    public IntakeSubsystem() {
+        intakeMotor = new TalonFX(0); 
+        intakeMotor.setNeutralMode(NeutralModeValue.Brake);
+        //intakeMotor.setInverted(true);
+        rightBeamBreak = new DigitalInput(0);
+        leftBeamBreak = new DigitalInput(0);
+    }
+
+    public void setMotorSpeed(double speed) {
+        intakeMotor.set(speed);
+    }
+
+    public boolean getRightBeamBreak() {
+        return rightBeamBreak.get();
+    }
+
+    public boolean getLeftBeamBreak() {
+        return leftBeamBreak.get();
```

Comment:
My only other question is: do we need to debounce beam breaks? I have no idea. Check with Sidd.

Reply:
I don't think so, I haven't seen much of a false positive rate with the Adafruit beam break sensors. But that might change with the other beam break methods that we use.